goprocess/gp.go

```diff
@@ -29,6 +29,15 @@ type P struct {
 func FindAll() []P {
 	var results []P
 
+	found := make(chan P)
```

Comment:
How about, let's remove results []P and then make found a channel like you have but no need for the done channel either since we already have a sync.WaitGroup below to ensure that all the results return
```diff
diff --git a/goprocess/gp.go b/goprocess/gp.go
index 9c211df..6b8a604 100644
--- a/goprocess/gp.go
+++ b/goprocess/gp.go
@@ -27,15 +27,14 @@ type P struct {
 
 // FindAll returns all the Go processes currently running on this host.
 func FindAll() []P {
-	var results []P
-
 	pss, err := ps.Processes()
 	if err != nil {
-		return results
+		return nil
 	}
 
 	var wg sync.WaitGroup
 	wg.Add(len(pss))
+	found := make(chan P)
 
 	for _, pr := range pss {
 		pr := pr
@@ -49,17 +48,26 @@ func FindAll() []P {
 			if !ok {
 				return
 			}
-			results = append(results, P{
+			found <- P{
 				PID:          pr.Pid(),
 				PPID:         pr.PPid(),
 				Exec:         pr.Executable(),
 				Path:         path,
 				BuildVersion: version,
 				Agent:        agent,
-			})
+			}
 		}()
 	}
-	wg.Wait()
+
+	go func() {
+		wg.Wait()
+		close(found)
+	}()
+
+	var results []P
+	for result := range found {
+		results = append(results, result)
+	}
 	return results
 }
```